yjit/src/codegen.rs

```diff
@@ -670,10 +657,11 @@ pub fn gen_entry_prologue(cb: &mut CodeBlock, iseq: IseqPtr, insn_idx: u32) -> O
 
     asm.compile(cb);
 
-    // Verify MAX_PROLOGUE_SIZE
-    assert!(cb.get_write_pos() - old_write_pos <= MAX_PROLOGUE_SIZE);
-
-    return Some(code_ptr);
+    if (cb.has_dropped_bytes()) {
+        None
+    } else {
+        Some(code_ptr)
+    }
```

Comment:
Does this mean that we will fail at generating an entry point if we are right at the page boundary?

Reply:
No, it should work fine for that case. `cb.has_dropped_bytes` means it has failed to write code. It's retried and set to `false` again by `cb.next_page()` when it steps into `page_end_reserve`. If it's right before `page_end_reserve`, `cb.has_capacity(1)` could be `false`, but `cb.has_dropped_bytes()` should be still true unless you ran out of all pages.

Reply:
No, in those cases we switch to the next page inside the assembler and the dropped bytes statues is reset.